packages/beacon-node/src/chain/opPools/attestationPool.ts

```diff
@@ -178,14 +182,34 @@ export class AttestationPool {
 /**
  * Aggregate a new contribution into `aggregate` mutating it
  */
-function aggregateAttestationInto(aggregate: AggregateFast, attestation: phase0.Attestation): InsertOutcome {
+function aggregateAttestationInto(aggregate: AggregateFast, attestation: allForks.Attestation): InsertOutcome {
   const bitIndex = attestation.aggregationBits.getSingleTrueBit();
 
   // Should never happen, attestations are verified against this exact condition before
   if (bitIndex === null) {
     throw Error("Invalid attestation not exactly one bit set");
   }
 
+  if ("committeeBits" in attestation && !("committeeBits" in aggregate)) {
+    throw Error("Attempt to aggregate electra attestation into phase0 attestation");
+  }
+
+  if (!("committeeBits" in attestation) && "committeeBits" in aggregate) {
```

Comment:
we don't need these validation unless we have to for the type assertion
in order to reach this code, attestation data should be the same hence the same type of attestation